lib/puppet/type/exec.rb

```diff
@@ -205,6 +209,12 @@ def sync
       end
     end
 
+    newparam(:pass_through_shell, :boolean => true, :parent => Puppet::Parameter::Boolean, :required_features => :supports_pass_through_shell) do
```

Comment:
Not a blocking comment, but I have a preference for something shorter like `:use_shell`